rotkehlchen/data/importer.py

```diff
@@ -1016,19 +1016,20 @@ def _consume_shapeshift_trade(self, csv_row: Dict[str, Any]) -> None:
             buy_asset = A_SAI
         if sold_asset == A_DAI and timestamp <= SAI_TIMESTAMP:
             sold_asset = A_SAI
-        trade = Trade(
-            timestamp=timestamp,
-            location=Location.SHAPESHIFT,
-            base_asset=buy_asset,
-            quote_asset=sold_asset,
-            trade_type=TradeType.BUY,
-            amount=buy_amount,
-            rate=Price(rate),
-            fee=Fee(fee),
-            fee_currency=buy_asset,  # Assumption that minerFee is denominated in outputCurrency
-            link='',
-            notes=notes,
-        )
+        if rate > ZERO:
```

Comment:
If rate <= ZERO here you will have an exception for undefined variable `trade` at line 1033. You should probably simply do:

```python
if rate <= ZERO:
    log.warning('unexpected data .. bla bla more info on entry')
    return
```

So return earlier and then add the trade.

Reply:
Thanks, copied this code and worded a log message.